compiler/rustc_span/src/lib.rs

```diff
@@ -1843,6 +1843,8 @@ impl StableSourceFileId {
 }
 
 impl SourceFile {
+    const MAX_FILE_SIZE: u32 = u32::MAX - 1;
```

Comment:
why the -1 here?

Reply:
See https://github.com/rust-lang/rust/issues/132862

Reply:
This PR is ``fail fast for files LARGER than 4 GiB``. 4GiB is 2^32B is (u32::MAX + 1) bytes, so U32::MAX is 4Gi - 1, and U32::MAX - 1 is 4Gi - 2. What is the purpose of a constant for 4GiB - 2B when preventing use of files larger than 4GiB? Is that not off by 2?

Reply:
Originally this PR was for enforcing that files of size 4 GiB and larger would not work, meaning the maximum file size would be 4 GiB - 1 B = `u32::MAX` bytes.

However, I then discovered #132862, so files of size 4 GiB - 1 B and larger would not work, so the maximum file size is actually 4 GiB - 2 B = `u32::MAX - 1`. So no the current value is correct.

I've just tested this again to make sure. Using the current stable and nightly version of `rustc`, I get the following results:
| File size (bytes) | stable result                            | nightly result  |
| ----------------- | ---------------------------------------- | --------------- |
| 4,294,967,294     | success                                  | success         |
| 4,294,967,295     | ICE (#132862)                            | instant failure |
| 4,294,967,296     | failure after reading the file to memory | instant failure |